challenge_12/python/challenge_12.py

```diff
@@ -0,0 +1,72 @@
+def compress(strr):
+    buff = []
+    compressed = ''
+
+    buff.append(strr[0])
+    for character in strr[1:]:
+        if character == buff[-1]:
+            buff.append(character)
+        elif character != buff[-1]:
+            if len(buff) > 4:
+                compressed += '{c}#{n}'.format(
+                        c=buff[-1],
+                        n=len(buff)
+                        )
+            else:
+                compressed += ''.join(buff)
+            buff = [character]
+    else:
+        if len(buff) > 4:
+            compressed += '{c}#{n}'.format(
+                    c=buff[-1],
+                    n=len(buff)
+                    )
+        else:
+            compressed += ''.join(buff)
+        buff = [character]
+    
+    return compressed
+
+def decompress(strr):
+    strr = strr.replace('#', '')
+    repeats = ''
+    decompressed = ''
+    NUMS = ('0', '1', '2', '3', '4', '5', '6', '7', '8', '9')
```

Comment:
It does!

![Repl.it compiler running this code...](http://i.imgur.com/IB4szQX.jpg)

Multi-digit numbers are appended to the variable `repeats` until they end (i.e. a character is parsed). This conditional is at line 37, `if character not in NUMS:`, which will run line 43, `repeats += character`.